modules/grel/src/test/java/com/google/refine/grel/GrelTests.java

```diff
@@ -261,6 +261,34 @@ public void testColumnDependencies() throws ParsingException {
         }
     }
 
+    @Test
+    public void testGetSource() throws ParsingException {
+        // integration test for getSource()
+        String tests[][] = {
+                { "value", "value" },
+                { "cell.recon.match.id", "cell.recon.match.id" },
+                { "value + 'a'", "value + \"a\"" },
+                { "\"foo\"", "\"foo\"" },
+                { "'\"'", "\"\\\"\"" }, // TODO we could print the string with the original quotes to avoid the escaping
+                { "1", "1" },
+                { "4 * (5 + 6)", "4 * (5 + 6)" },
+                { "cells.foo", "cells.foo" },
+                { "value + ' ' + cells.foo.value", "value + \" \" + cells.foo.value" },
+                { "cells[\"foo\"].value", "cells.get(\"foo\").value" }, // TODO this could be more faithful
+                { "toDate( value+4 )", "toDate(value + 4)" },
+                { "(value + 4).toDate()", "(value + 4).toDate()" }, // TODO this could potentially be improved
+                { "forEach([3, 4,8,7], v, mod(v,2))", "forEach([3, 4, 8, 7], v, mod(v, 2))" },
```

Comment:
This test gives an overview of the sorts of differences introduced by the pretty-printing logic.